src/main/java/com/cleanroommc/groovyscript/registry/AbstractCraftingRecipeBuilder.java

```diff
@@ -168,6 +170,12 @@ protected <T> T validateShape(GroovyLog.Msg msg, List<String> errors, String[] k
         }
         int finalRowWidth = rowWidth;
         msg.add(rowWidth > width, () -> "At least one row has a row length of " + finalRowWidth + ", but maximum is " + width);
+
+        if (GroovyScriptConfig.compat.checkInputStackCounts) {
+            int maxAmountProvided = ingredients.stream().mapToInt(IResourceStack::getAmount).max().orElse(0);
+            msg.add(maxAmountProvided > 1, "All crafting slot inputs must have a size of 1, got {}", maxAmountProvided);
```

Comment:
for the sake of parity and the possibility that there are some crafting tables which can have multiple items in a stack, can this also be made into a method?

Reply:
Yeah I did consider it, will add one.